plasmapy/dispersion/numerical/hollweg_.py

```diff
@@ -0,0 +1,352 @@
+    >>> inputs = {
+    ...    "k": np.logspace(-7, -2, 2) * u.rad / u.m,
+    ...    "theta": 88 * u.deg,
+    ...    "n_i": 5 * u.cm ** -3,
+    ...    "B": 2.2e-8 * u.T,
+    ...    "T_e": 1.6e6 * u.K,
+    ...    "T_i": 4.0e5 * u.K,
+    ...    "ion": Particle("p+"),
+    ... }
+    >>> omegas = hollweg(**inputs)
+    >>> omegas
+    {'fast_mode': <Quantity [2.62911663e-02+0.j, 2.27876968e+03+0.j] rad / s>,
+     'alfven_mode': <Quantity [7.48765909e-04+0.j, 2.13800404e+03+0.j] rad / s>,
+     'acoustic_mode': <Quantity [0.00043295+0.j, 0.07358991+0.j] rad / s>}
+    """
+
+    # validate argument ion
+    if not isinstance(ion, Particle):
+        try:
+            ion = Particle(ion)
+        except TypeError:
+            raise TypeError(
+                f"For argument 'ion' expected type {Particle} but got {type(ion)}."
+            )
+    if not (ion.is_ion or ion.is_category("element")):
+        raise ValueError("The particle passed for 'ion' must be an ion or element.")
+
+    # validate z_mean
+    if z_mean is None:
+        try:
+            z_mean = abs(ion.charge_number)
+        except ChargeError:
+            z_mean = 1
+    else:
+        if not isinstance(z_mean, (int, np.integer, float, np.floating)):
+            raise TypeError(
+                f"Expected int or float for argument 'z_mean', but got {type(z_mean)}."
+            )
+        z_mean = abs(z_mean)
+
+    # validate arguments
+    for arg_name in ("B", "n_i", "T_e", "T_i"):
+        val = locals()[arg_name].squeeze()
+        if val.shape != ():
+            raise ValueError(
+                f"Argument '{arg_name}' must be single valued and not an array of "
+                f"shape {val.shape}."
+            )
+        locals()[arg_name] = val
+
+    # validate arguments
+    for arg_name in ("gamma_e", "gamma_i"):
+        if not isinstance(locals()[arg_name], (int, np.integer, float, np.floating)):
+            raise TypeError(
+                f"Expected int or float for argument '{arg_name}', but got "
+                f"{type(locals()[arg_name])}."
+            )
+
+    # validate argument k
+    k = k.squeeze()
+    if not (k.ndim == 0 or k.ndim == 1):
+        raise ValueError(
+            f"Argument 'k' needs to be single valued or a 1D array astropy Quantity,"
+            f" got array of shape {k.shape}."
+        )
+    if np.any(k <= 0):
+        raise ValueError("Argument 'k' can not be a or have negative values.")
+
+    # validate argument theta
+    theta = theta.squeeze()
+    if not (theta.ndim == 0 or theta.ndim == 1):
+        raise ValueError(
+            f"Argument 'theta' needs to be a single valued or 1D array astropy "
+            f"Quantity, got array of shape {k.shape}."
+        )
+
+    # Single k value case
+    if np.isscalar(k.value):
+        k = (
+            np.array(
+                [
+                    k.value,
+                ]
+            )
+            * u.rad
+            / u.m
+        )
```

Comment:
```suggestion
    if np.isscalar(k.value):
        k = np.array([k.value]) * u.rad / u.m
```

A cleaner way that does not make `black` upset.  I always forget that having the comma `[k.value,]` causes `black` to want to split everything onto its own line.